### PR TITLE
[FIX] account_edi: don't allow to change in sequence

### PR DESCRIPTION
**Pull request by zepa-odoo**
Before PR:
---
If the invoice has already been sent to the government, then it's allowed to change in sequence.

After PR:
---
The sequence can't be changed once the invoice is created and sent to the government.

task ID :- 3254322

### CODE REVIEW

File: addons/account_edi/wizard/account_resequence.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+from odoo import _, models
+from odoo.exceptions import UserError
+
+class ReSequenceWizard(models.TransientModel):
+    _inherit = 'account.resequence.wizard'
+
+    def resequence(self):
+        edi_is_send = self.move_ids.edi_document_ids.filtered(lambda d: d.edi_format_id._needs_web_services() and d.state in ('sent'))
```
Comment (RicGR98) [inline addons/account_edi/wizard/account_resequence.py:9]: ~~`is_sent` instead of `is_send`~~

Actually it's the recordset of moves which were sent, so I'd use `edi_sent_moves` instead so it fits better the following usages in the code

Comment (RicGR98) [inline addons/account_edi/wizard/account_resequence.py:9]: `d.state in ('sent')`
You're only checking one state, so I guess an equality would more appropriate than an `in` ?

File: addons/account_edi/wizard/account_resequence.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        if edi_is_send:
+            move_types_name = " and ".join(set(edi_is_send.move_id.mapped('type_name')))
+            service_types_name = " and ".join(set(edi_is_send.edi_format_id.mapped('name')))
+            raise UserError(_(
+                "Some %s has already sent to %s, so you can't change sequence.\n"\
+                "You can cancel it and create %s with new sequence.", move_types_name, service_types_name, move_types_name))
```
Comment (RicGR98) [inline addons/account_edi/wizard/account_resequence.py:15]: I would rephrase this (as well as the pot) as follows:

```suggestion
            raise UserError(_(
                "The following documents have already been sent and cannot be resequenced: ",
                ", ".join(set(edi_sent_moves.move_id.mapped('name')))
            ))
```
What do you think ?

**Review by RicGR98 — COMMENTED** [review]
Thanks for the PR !

File: addons/account_edi/wizard/account_resequence.py (hunk 2 above)
Comment (zepa-odoo) [reply addons/account_edi/wizard/account_resequence.py]: Thank you @RicGR98 for suggesting this change.

**Comment by RicGR98** [comment]
Thanks for the correction ! :)

Once it's functionally validated by a PO, I'll forward it to the final reviewer

**Comment by zepa-odoo** [comment]
This PR closed due to the task being canceled... 😢

File: addons/account_edi/wizard/account_resequence.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+class ReSequenceWizard(models.TransientModel):
+    _inherit = 'account.resequence.wizard'
+
+    def resequence(self):
+        edi_sent_moves = self.move_ids.edi_document_ids.filtered(lambda d: d.edi_format_id._needs_web_services() and d.state == 'sent')
+        if edi_sent_moves:
+            raise UserError(_("The following documents have already been sent and cannot be resequenced: %s")
+                % ", ".join(set(edi_sent_moves.move_id.mapped('name')))
+            )
+        super().resequence()
```
Comment (RicGR98) [inline addons/account_edi/wizard/account_resequence.py:14]:
```suggestion
        return super().resequence()
```

Even though the parent method does not return anything currently, it may one day, so I don't think this would hurt ^^

File: addons/account_edi/wizard/account_resequence.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
```
Comment (RicGR98) [inline addons/account_edi/wizard/account_resequence.py:1]:
```suggestion
```

File: addons/account_edi/wizard/__init__.py [hunk 5]
```diff
@@ -0,0 +1,2 @@
+# -*- encoding: utf-8 -*-
```
Comment (RicGR98) [inline addons/account_edi/wizard/__init__.py:1]:
```suggestion
```

**Review by RicGR98 — COMMENTED** [review]
Thanks for the PR :)

**Review by zepa-odoo — COMMENTED** [review]
Hello @RicGR98,
I've Done all the suggested changes.

**Review by RicGR98 — APPROVED** [review]
Thanks for the changes !

I'll forward it to the final reviewer :)

**Review by oco-odoo — APPROVED** [review]
@robodoo r+
